cloudinit/log.py

```diff
@@ -23,6 +23,23 @@
 
 DEFAULT_LOG_FORMAT = "%(asctime)s - %(filename)s[%(levelname)s]: %(message)s"
 DEPRECATED = 35
+TRACE = logging.DEBUG - 5
+
+
+class CustomLoggerType(logging.Logger):
+    """A hack to get mypy to stop complaining about custom logging methods.
+
+    When using deprecated or trace logging, rather than:
+        LOG = logging.getLogger(__name__)
+    Instead do:
+        LOG = cast(CustomLoggerType, logging.getLogger(__name__))
+    """
+
+    def trace(self, *args, **kwargs):
```

Comment:
Think we should document our custom loggers in the dev docs?

I think that describing when one should choose debug vs choose trace would help guide the project to use these in standard ways.

Reply:
Good idea. Since we have a docs sprint coming up, I filed https://github.com/canonical/cloud-init/issues/5525 and hopefully we can address it then.